prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/Prometheus.scala

```diff
@@ -191,6 +193,60 @@ object Prometheus {
   private val DefaultHistogramBuckets: NonEmptyList[Double] =
     NonEmptyList(.005, List(.01, .025, .05, .075, .1, .25, .5, .75, 1, 2.5, 5, 7.5, 10))
 
+  /**
+    * Given an exclude function, return a 'classifier' function, i.e. for application in
+    * [[org.http4s.server/client.middleware.Metrics#apply]].
+    *
+    * Let's say you want a classifier that excludes integers since your paths consist of:
+    *   * GET    /users/{integer} = get_users_*
+    *   * POST   /users           = post_users
+    *   * PUT    /users/{integer} = put_users_*
+    *   * DELETE /users/{integer} = delete_users_*
+    *
+    * In such a case, we could use:
+    *
+    * classifierFMethodWithOptionallyExcludedPath { str: String => scala.util.Try(str.toInt).isSuccess }
+    *
+    * @param exclude For a given String, namely a path value, determine whether the value gets excluded.
+    * @return Request[F] => Option[String]
+    */
+  def classifierFMethodWithOptionallyExcludedPath[F[_]](
+      exclude: String => Boolean
+  ): Request[F] => Option[String] = { request: Request[F] =>
+    val initial: String = request.method.name.toLowerCase
+
+    val pathList: List[String] =
+      requestToPathList(request)
+
+    val minusExcluded: List[String] = pathList.map { value: String =>
+      if (exclude(value)) "*" else value.toLowerCase
+    }
+
+    val result: String =
+      minusExcluded match {
+        case Nil => initial
+        case nonEmpty @ _ :: _ => initial + "_" + Foldable[List].intercalate(nonEmpty, "_")
+      }
+
+    Some(result)
+  }
+
+  // The following was copied from
+  // https://github.com/http4s/http4s/blob/v0.20.17/dsl/src/main/scala/org/http4s/dsl/impl/Path.scala#L56-L64,
+  // and then modified.
+  private def requestToPathList[F[_]](request: Request[F]): List[String] = {
+    val str: String = request.pathInfo
+
+    if (str == "" || str == "/")
+      Nil
+    else {
+      val segments = str.split("/", -1)
+      // .head is safe because split always returns non-empty array
+      val segments0 = if (segments.head == "") segments.drop(1) else segments
+      segments0.foldLeft[List[String]](Nil)((path, seg) => path ++ List(Uri.decode(seg)))
```

Comment:
This method is super inefficient, we should fix it in both places to cons and then reverse afterwards.

Reply:
ACK - https://github.com/http4s/http4s/pull/3167/commits/03517004f7b22cd5b5f9b4cb70a1e9d1239ef187.